src/dbt_client/dbtClientCommandQueue.ts

```diff
@@ -0,0 +1,36 @@
+import { StatusBarAlignment, StatusBarItem, window } from "vscode";
+
+type Command = () => Promise<any>;
+
+class DBTClientCommandQueue {
+    private queue: Command[] = [];
+    private currentPromise: any;
+    private statusBar: StatusBarItem;
+
+    constructor() {
```

Comment:
I'm not sure the statusbar should be integrated in this class.

I think you can always show the status bar and show the version of the installed DBT, and when it's working you update what you are running exactly, for example '<spinning wheel> listing dbt models...' or '<spinning wheel> running model(s)...'